runtime/compiler/runtime/JITServerAOTCache.cpp

```diff
@@ -65,6 +79,32 @@ AOTCacheClassLoaderRecord::create(uintptr_t id, const uint8_t *name, size_t name
    return new (ptr) AOTCacheClassLoaderRecord(id, name, nameLength);
    }
 
+AOTCacheClassLoaderRecord *
+AOTCacheClassLoaderRecord::read(FILE *f,
+                                const Vector<AOTCacheClassLoaderRecord *> &classLoaderRecords,
+                                const Vector<AOTCacheClassRecord *> &classRecords,
+                                const Vector<AOTCacheMethodRecord *> &methodRecords,
+                                const Vector<AOTCacheClassChainRecord *> &classChainRecords,
+                                const Vector<AOTCacheWellKnownClassesRecord *> &wellKnownClassesRecords,
+                                const Vector<AOTCacheAOTHeaderRecord *> &aotHeaderRecords)
+   {
+   ClassLoaderSerializationRecord header;
+   if (1 != fread(&header, sizeof(header), 1, f))
+      return NULL;
+
+   if (!header.isValid())
+      return NULL;
+
+   auto record = new (AOTCacheRecord::allocate(size(header.nameLength()))) AOTCacheClassLoaderRecord();
+   memcpy((void *)record->dataAddr(), &header, sizeof(header));
+   if (1 != fread((uint8_t *)record->dataAddr() + sizeof(header), (static_cast<AOTSerializationRecord&>(header)).size() - sizeof(header), 1, f))
```

Comment:
Why do you need the `static_cast`? `ClassLoaderSerializationRecord` is a subclass of `AOTSerializationRecord`, it inherits the `size()` method.

Reply:
It was to disambiguate between `AOTSerializationRecord::size` and the static `ClassLoaderSerializationRecord::size`, but like you pointed out above I can just call `AOTSerializationRecord::size` directly.